src/test/java/org/terasology/logic/inventory/InventoryManagerTest.java

```diff
@@ -0,0 +1,207 @@
+
+public class InventoryManagerTest {
+
+    static final String URI_DIRT = "CoreAssets:Dirt";
+
+    private static ModuleTestingEnvironment context;
+
+    private Context hostContext;
+    private EntityManager entityManager;
+    private InventoryManager inventoryManager;
+    private BlockManager blockManager;
+    private EntityRef inventory;
+
+    @BeforeAll
+    public static void setup() throws Exception {
+        context = new ModuleTestingEnvironment() {
+            @Override
+            public Set<String> getDependencies() {
+                return Sets.newHashSet("CoreAssets", "Inventory");
+            }
+        };
+        context.setup();
+    }
+
+    @AfterAll
+    public static void tearDown() throws Exception {
+        context.tearDown();
+    }
+
+    @BeforeEach
+    public void beforeEach() {
+        hostContext = context.getHostContext();
+
+        entityManager = hostContext.get(EntityManager.class);
+        inventoryManager = hostContext.get(InventoryManager.class);
+        blockManager = hostContext.get(BlockManager.class);
+
+        inventory = entityManager.create(new InventoryComponent(3));
+    }
+
+    @ParameterizedTest
+    @ValueSource(bytes = {0, -1})
+    public void giveItem_zeroBlocks(int amount) {
+        EntityRef blockItem = getDirtBlockItem(entityManager, blockManager, amount);
+        inventoryManager.giveItem(inventory, EntityRef.NULL, blockItem);
+
+        final List<EntityRef> filledSlots = getFilledSlots(inventory);
+        Assert.assertEquals("No slots should be filled", 0, filledSlots.size());
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = {1, 99})
+    public void giveItem_singleStack(int amount) {
+        EntityRef blockItem = getDirtBlockItem(entityManager, blockManager, amount);
+        inventoryManager.giveItem(inventory, EntityRef.NULL, blockItem);
+
+        final List<EntityRef> filledSlots = getFilledSlots(inventory);
+        Assert.assertEquals("Exactly one slot should be filled", 1, filledSlots.size());
+        filledSlots.forEach(item -> {
+            Assert.assertEquals("Slot should contain exactly the added block", blockItem, item);
+
+            final int maxStackSize = blockItem.getComponent(ItemComponent.class).maxStackSize;
+            Assert.assertEquals("Stack count should be " + amount + " or maximum stack size",
+                    Math.min(amount, maxStackSize), item.getComponent(ItemComponent.class).stackCount);
+        });
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = {100, 128, 256})
+    public void giveItem_blockExceedingStacksize(int amount) {
+        EntityRef blockItem = getDirtBlockItem(entityManager, blockManager, amount);
+        inventoryManager.giveItem(inventory, EntityRef.NULL, blockItem);
+
+        final List<EntityRef> filledSlots = getFilledSlots(inventory);
+        Assert.assertEquals("Exactly one slot should be filled", 1, filledSlots.size());
+        filledSlots.forEach(item -> {
```

Comment:
We junit 5's `Assetions:
```
Assertions.assertAll(
    filledSlots.stream()
                    .map((item)-> ()-> { // Lambda of Executable - Consumer with throws exception.
                        /// Your assertions.
   });
```
Also, you can use faltMap, if you want.

Reply:
Good point, but became a bit lengthy ....

```java
        Assertions.assertAll(
                filledSlots.stream().flatMap(item ->
                        Stream.of(
                                () -> Assertions.assertEquals(blockItem, item,
                                        "Slot should contain exactly the added block"),
                                () -> Assertions.assertEquals(amount, item.getComponent(ItemComponent.class).stackCount,
                                        "Unexpected slot count!")
                        )
                )
        );
```

I'm pretty sure I'll use that at some point, though ^^